### ISSUE TITLE
Add some handy timing functions for RetryMiddleware

### ISSUE THREAD

**Issue by Thinkscape** [guzzle/guzzle#1213]
It seems that `RetryFirmware` replaces retry-subscriber (and other backoff-related classes from Guzzle < 6.). The issue is that there are no timing and "decider" functions provided out of the box. Good thing is, it's probably trivial to implement.

This ticket is meant to track progress on adding those, so users get the basic functionality of backoff and retry in Guzzle 6.

**Comment by mtdowling** [comment]
Yeah, that would be cool!

**Comment by gunnarlium** [comment]
I just ported an app from Guzzle 5 to 6, which needs retry functionality. The retry part itself is really quite easy to setup, however, I had to read a lot of code to figure out how. I think maybe this is best solved my adding a cookbook or something for how to implement your own retry. As the logic is likely to vary depending on different apps, I expect a simple recipe might be better than a full default implementation.
